bridges/snowbridge/primitives/inbound-queue/src/v2/converter.rs

```diff
@@ -0,0 +1,624 @@
+	InboundQueuePalletInstance: Get<u8>,
+	ConvertAssetId: MaybeEquivalence<TokenId, Location>,
+	GatewayProxyAddress: Get<H160>,
+	EthereumUniversalLocation: Get<InteriorLocation>,
+	GlobalAssetHubLocation: Get<Location>,
+{
+	_phantom: PhantomData<(
+		EthereumNetwork,
+		InboundQueuePalletInstance,
+		ConvertAssetId,
+		GatewayProxyAddress,
+		EthereumUniversalLocation,
+		GlobalAssetHubLocation,
+	)>,
+}
+
+impl<
+		EthereumNetwork,
+		InboundQueuePalletInstance,
+		ConvertAssetId,
+		GatewayProxyAddress,
+		EthereumUniversalLocation,
+		GlobalAssetHubLocation,
+	> ConvertMessage
+	for MessageToXcm<
+		EthereumNetwork,
+		InboundQueuePalletInstance,
+		ConvertAssetId,
+		GatewayProxyAddress,
+		EthereumUniversalLocation,
+		GlobalAssetHubLocation,
+	>
+where
+	EthereumNetwork: Get<NetworkId>,
+	InboundQueuePalletInstance: Get<u8>,
+	ConvertAssetId: MaybeEquivalence<TokenId, Location>,
+	GatewayProxyAddress: Get<H160>,
+	EthereumUniversalLocation: Get<InteriorLocation>,
+	GlobalAssetHubLocation: Get<Location>,
+{
+	fn convert(
+		message: Message,
+	) -> Result<Xcm<()>, ConvertMessageError> {
+		let mut message_xcm: Xcm<()> = Xcm::new();
+		if message.xcm.len() > 0 {
+			let xcm_string = hex::encode(message.xcm.clone());
+			log::info!(target: LOG_TARGET,"found xcm payload: {:x?}", xcm_string);
+			// Allow xcm decode failure so that assets can be trapped on AH instead of this
+			// message failing but funds are already locked on Ethereum.
+			if let Ok(versioned_xcm) = VersionedXcm::<()>::decode_with_depth_limit(
+				MAX_XCM_DECODE_DEPTH,
+				&mut message.xcm.as_ref(),
+			) {
+				if let Ok(decoded_xcm) = versioned_xcm.try_into() {
+					message_xcm = decoded_xcm;
+				} else {
+					log::error!(target: LOG_TARGET,"unable to decode xcm");
+				}
+			} else {
+				log::error!(target: LOG_TARGET,"unable to decode versioned xcm");
+			}
+		}
+
+		log::info!(target: LOG_TARGET,"xcm decoded as {:?}", message_xcm);
+
+		let network = EthereumNetwork::get();
+
+		// use eth as asset
+		let fee_asset = Location::new(2, [GlobalConsensus(EthereumNetwork::get())]);
+		let fee: Asset = (fee_asset.clone(), message.execution_fee).into();
+		let eth: Asset =
+			(fee_asset.clone(), message.execution_fee.saturating_add(message.value)).into();
+		let mut instructions = vec![
+			DescendOrigin(PalletInstance(InboundQueuePalletInstance::get()).into()),
+			UniversalOrigin(GlobalConsensus(network)),
+			ReserveAssetDeposited(eth.into()),
+			PayFees { asset: fee },
+		];
+		let mut reserve_assets = vec![];
+		let mut withdraw_assets = vec![];
+
+		// Let origin account transact on AH directly to reclaim assets and surplus fees.
+		// This will be possible when AH gets full EVM support
+		let default_claimer = Location::new(0, [
+			AccountKey20 {
+				// Set network to `None` to support future Plaza EVM chainid by default.
+				network: None,
+				// Ethereum account ID
+				key: message.origin.as_fixed_bytes().clone()
+			}
+		]);
```

Comment:
@vgeddes so we can't really have this as a default claimer, right? Other options for the default claimer are (if the UX does not provide a claimer):

- The first beneficiary of a user added DepositAsset
- The relayer

Reply:
@vgeddes let me know what you think of this. Currently the register token v2 token smoke tests fails because of the Account20 claimer added to the message.

Reply:
Deposit to Snowbridge sovereign.